### PR TITLE
DOC: Move API breaking to appropriate sections

### PR DESCRIPTION
**Pull request by TomAugspurger**
As discussed in https://github.com/pandas-dev/pandas/issues/34801, this moves some items from our "API breaking changes" section to either bug fixes or enhancements. I've only moved the ones we can hopefully get consensus on as bug fix / enhancement. But let me know if any of these need further discussion on whether they're an API breaking change.

### CODE REVIEW

File: doc/source/whatsnew/v1.1.0.rst [hunk 1]
```diff
@@ -902,6 +767,94 @@ Interval
 
 Indexing
 ^^^^^^^^
+
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:770]: hmm you really want this in the middle of the bug fix section?

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.1.0.rst (hunk 1 above)
Comment (TomAugspurger) [reply doc/source/whatsnew/v1.1.0.rst:770]: I can shorten these to one-sentence descriptions and ensure that the linked issue has a clear example. Would that be preferable?

File: doc/source/whatsnew/v1.1.0.rst [hunk 2]
```diff
@@ -623,57 +532,14 @@ The method :meth:`core.DataFrameGroupBy.size` would previously ignore ``as_index
 
    df.groupby("a", as_index=False).size()
 
-.. _whatsnew_110.api_breaking.apply_applymap_first_once:
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:626]: where did this one go?

File: doc/source/whatsnew/v1.1.0.rst [hunk 3]
```diff
@@ -524,39 +466,6 @@ those integer keys is not present in the first level of the index (:issue:`33539
 
     left_df.merge(right_df, on=['animal', 'max_speed'], how="right")
 
-.. ---------------------------------------------------------------------------
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:527]: where did this go?

File: doc/source/whatsnew/v1.1.0.rst [hunk 4]
```diff
@@ -336,68 +340,6 @@ Other enhancements
 Backwards incompatible API changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-``MultiIndex.get_indexer`` interprets `method` argument differently
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:339]: where did this go?

**Review by jreback — CHANGES_REQUESTED** [review]
unless i am missing something a fair amount was deleted here

**Comment by TomAugspurger** [comment]
As requested in https://github.com/pandas-dev/pandas/pull/34865/files/a1130882c28ac17be013287fd5abef40841e1d8e#diff-e763086372257ed7ca944d8bc5042a04

File: doc/source/whatsnew/v1.1.0.rst (hunk 4 above)
Comment (TomAugspurger) [reply doc/source/whatsnew/v1.1.0.rst:339]: https://github.com/pandas-dev/pandas/pull/34865/files#diff-e763086372257ed7ca944d8bc5042a04R813

File: doc/source/whatsnew/v1.1.0.rst (hunk 3 above)
Comment (TomAugspurger) [reply doc/source/whatsnew/v1.1.0.rst:527]: https://github.com/pandas-dev/pandas/pull/34865/files#diff-e763086372257ed7ca944d8bc5042a04R812

File: doc/source/whatsnew/v1.1.0.rst (hunk 2 above)
Comment (TomAugspurger) [reply doc/source/whatsnew/v1.1.0.rst:626]: https://github.com/pandas-dev/pandas/pull/34865/files#diff-e763086372257ed7ca944d8bc5042a04R1002

**Review by TomAugspurger — COMMENTED** [review]
Would be nice to get this in before more merge conflicts come up.

**Comment by TomAugspurger** [comment]
Also cc @jorisvandenbossche for confirmation that this all look like bug fixes.

File: doc/source/whatsnew/v1.1.0.rst (hunk 4 above)
Comment (jreback) [reply doc/source/whatsnew/v1.1.0.rst:340]: I would rather not move this, this example is very needed to understand what the heck that change was.

I have no problem with you moving things around, but let's leave the examples. If you want to remove them, then let's discuss individually.

**Review by jreback — CHANGES_REQUESTED** [review]
see my comments. let's move things around, leaving the examples. and potentially change / remove them later (as to be honest I am not sure why you are deleting them in the first place).

**Comment by TomAugspurger** [comment]
I need to close this for a while.

**Pull request closed without merging**